CIScripts/Tests/TestConfigurationUtils.ps1

```diff
@@ -0,0 +1,174 @@
+}
+
+class TestConfiguration {
+    [DockerDriverConfiguration] $DockerDriverConfiguration;
+    [string] $AdapterName;
+    [string] $VMSwitchName;
+    [string] $ForwardingExtensionName;
+}
+
+function Enable-VRouterExtension {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [string] $AdapterName,
+           [Parameter(Mandatory = $true)] [string] $VMSwitchName,
+           [Parameter(Mandatory = $true)] [string] $ForwardingExtensionName)
+
+    Write-Host "Enabling Extension"
+
+    Invoke-Command -Session $sess -ScriptBlock {
+        New-ContainerNetwork -Mode Transparent -NetworkAdapterName $Using:AdapterName -Name test_network
+        Enable-VMSwitchExtension -VMSwitchName $Using:VMSwitchName -Name $Using:ForwardingExtensionName | Out-Null
+    }
+}
+
+function Disable-VRouterExtension {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [string] $VMSwitchName,
+           [Parameter(Mandatory = $true)] [string] $ForwardingExtensionName)
+
+    Write-Host "Disabling Extension"
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        Disable-VMSwitchExtension -VMSwitchName $Using:VMSwitchName -Name $Using:ForwardingExtensionName -ErrorAction SilentlyContinue | Out-Null
+        Get-ContainerNetwork | Remove-ContainerNetwork -Force
+    }
+}
+
+function Test-VRouterExtensionEnabled {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [string] $VMSwitchName,
+           [Parameter(Mandatory = $true)] [string] $ForwardingExtensionName)
+
+    $Ext = Invoke-Command -Session $Session -ScriptBlock {
+        return $(Get-VMSwitchExtension -VMSwitchName $Using:VMSwitchName -Name $Using:ForwardingExtensionName -ErrorAction SilentlyContinue)
+    }
+
+    return $($Ext.Enabled -and $Ext.Running)
+}
+
+function Enable-DockerDriver {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [string] $AdapterName,
+           [Parameter(Mandatory = $true)] [DockerDriverConfiguration] $Configuration)
+
+    Write-Host "Enabling Docker Driver"
+
+    $TenantName = $Configuration.NetworkConfiguration.TenantName
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        # Nested ScriptBlock variable passing workaround
+        $AdapterName = $Using:AdapterName
+        $Configuration = $Using:Configuration
+        $TenantName = $Using:TenantName
+
+        Start-Job -ScriptBlock {
+            $Configuration = $Using:Configuration
+
+            $Env:OS_USERNAME = $Configuration.Username
+            $Env:OS_PASSWORD = $Configuration.Password
+            $Env:OS_AUTH_URL = $Configuration.AuthUrl
+            $Env:OS_TENANT_NAME = $Using:TenantName
```

Comment:
`Configuration` is a local variable because of objects passing workaround, but OK, it's a good idea.